tests/topotests/multicast_pim6_dual_stack/test_multicast_pim6_dual_stack.py

```diff
@@ -0,0 +1,1936 @@
+
+    step(
+        "IPv4 and IPv6 traffic updated per (S,G)"
+        "show ipv6 multicast count ,show ip multicast count"
+    )
+
+    step("verify ipv6 multicast traffic on all the grps")
+    result = verify_sg_traffic(tgen, "r1", MLD_JOIN_RANGE_1, source_v6, "ipv6")
+    assert result is True, "Testcase {} : Failed Error: {}".format(tc_name, result)
+
+    step("verify ipv4 multicast traffic on all the grps")
+    result = verify_sg_traffic(tgen, "r1", IGMP_JOIN_RANGE_1, source_v4, "ipv4")
+    assert result is True, "Testcase {} : Failed Error: {}".format(tc_name, result)
+
+    step(
+        "Verify IP pimv6 join and IPv6 pimv6 join to FRR1 and check join is sent toward FHR"
+        "Join counts is increamented for both IPv4 and IPv6"
+        'verify using "show ip pimv6 intrface traffic" "show ipv6 pimv6 interface traffic"'
+    )
+
+    state_after_v6 = verify_pim_interface_traffic(tgen, state_dict, addr_type="ipv6")
+    assert isinstance(
+        state_after_v6, dict
+    ), "Testcase {} : Failed \n state_before is not dictionary \n " "Error: {}".format(
+        tc_name, result
+    )
+
+    result = verify_state_incremented(state_before_v6, state_after_v6)
+    assert result is True, "Testcase {} : Failed Error: {}".format(tc_name, result)
+
+    state_after_v4 = verify_pim_interface_traffic(tgen, state_dict, addr_type="ipv4")
+    assert isinstance(
+        state_after_v4, dict
+    ), "Testcase {} : Failed \n state_before is not dictionary \n " "Error: {}".format(
+        tc_name, result
+    )
+
+    result = verify_state_incremented(state_before_v4, state_after_v4)
+    assert result is True, "Testcase {} : Failed Error: {}".format(tc_name, result)
+
+    write_test_footer(tc_name)
+
+
+def test_ipv4_ipv6_mroutes_upstream_when_mld_and_igmp_join_sent_from_different_interface_p0(
+    request, app_helper
+):
+    """
+    Verify IPv4 and IPv6 mroute and upstream when mld and MLD join sent from different interface
+    of FRR1 and source present on different interface in FRR4
+    """
+
+    tgen = get_topogen()
+    tc_name = request.node.name
+    write_test_header(tc_name)
+    app_helper.stop_all_hosts()
+
+    # Creating configuration from JSON
+    reset_config_on_routers(tgen)
+
+    # Don"t run this test if we have any failure.
+    if tgen.routers_have_failure():
+        pytest.skip(tgen.errors)
+
+    step("Done in base config: " "Configure EBGP peering between all the nodes")
+
+    step("Done in base config: " "Enable PIM6 on all the interfaces of all the nodes")
+
+    step("Enable MLD on receiver interface")
+    intf_r1_i1 = topo["routers"]["r1"]["links"]["i1"]["interface"]
+    input_dict_1 = {
+        "r1": {"mld": {"interfaces": {intf_r1_i1: {"mld": {"version": "1"}}}}}
+    }
+    result = create_mld_config(tgen, topo, input_dict_1)
+    assert result is True, "Testcase {}: Failed Error: {}".format(tc_name, result)
+
+    step("Enable IGMP on receiver interface on same interface")
```

Comment:
the api name says igmp and mld on different interface but the comment says here same interface, can you please correct it.